Reorganize test folders

ranechita opened the pull request:
With the way tests are currently organized in folders, the code in setup_env.sh, config.sh, lib/production.sh, etc was mostly duplicated in every production test folder, with very few additions and modifications. All the production scripts were moved to a single folder main_tests, which contains all the common files and device specific folders only for scripts that are particular for a certain setup. 

The setup_env script is run with a BOARD parameter, based on which the pi is configured specifically for the chosen board.
The changes were tested with 2 different setups (synchrona and adrv9361) and the overall functionality and user experience is the same.

Waiting for suggestions for naming or any further structure related changes.

AlexandraTrifan committed:
synchrona: Add initial version for the M2k script that determines the
frequency and peaks for a measured signal.

Signed-off-by: Alexandra Trifan <Alexandra.Trifan@analog.com>

ranechita committed:
Add production tests for adrv1crr

Add production tests similar to existing ones,
catered for adrv1crr.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
test_prod: adrv1crr: update nav test

Nav test checking for appropriate results from
buttons, switches and leds.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: adrv1crr: periph + gpio

Add appropriate offsets and tests for GPIOs
and peripherals.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: remove test_hmc

HMC was tested for talise, irrelevant for ADRVCRR1.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: usb host mode

Add functional tests for USB host mode.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: audio test

Modify commands for appropriately testing audio
loopback.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: adrv1crr: clocks

Test that all clocks have appropriate frequency.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: adrv1crr: fix naming and beautify

Change naming for board and tests and echo some
extra information regarding number of tests passed

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: adrv1crr: update pass/fail conditions

Test now fails if even just 1 test fails. Updated echos

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: adrv1crr: add usb peripheral test

Add appropriate test for the usb in device mode.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: adrv1crr: test fixes and updates

Fix small bugs in pass/fail conditions
and uart_expect message. Detect button events
instantaneously, check for date-time
synchronization on pi.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: adrv1crr: add board serial number

Check for carrier serial number instead of SOM.
Input can be from scanner.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
scanner

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_test: fmcomms4: initial commit

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: fmcomms4: separate rf from dcxo

DCXO test is listed as separate option.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod_tests: fmcomms4: structure fmcomms4 tests

add libraries installation at setup, adapt paths
and calls to functions, etc.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: add initial output tests

Test output frequencies one by one.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: Add test for back pannel inputs

Added scripts for testing inputs of hmc and
ad9545.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: adapt to dtoverlay changes

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: add gpio tests

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: move test files to folder synch

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: add setup files

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: adrv1crr: fixes around otg on older revisions

On some SOM boards the otg usb doesn't
switch to host mode automatically.
Before testing the otg funcrionality an rebind of
the platform driver is performed so that the
device responds accordingly.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: add misc tests and production script

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: add synchrona general script

Added one script to call all the other
which run on synchrona.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: minor tests fixes

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: add support for printing labels

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: minor test updates

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: add overlays in test folder

Added the overlays for the test cases in the test folder.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: fix label path

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: eeprom writing attempt2

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: minor fixes and updates

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: fix writing to eeprom

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: display the temperature before checking intervals

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: adjust writing in the eeprom

Changed writing address 0x7ffe0 to ensure enough space is left
and used latest bootloader version.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synch: log more data for back pannel test

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synch: add manufacturing date to the labels

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synch: write in rpi eeprom using fru-dump

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synch: test most outputs as cmos

In order to verify N and P separately for each output
we configure the rest of the outputs as CMOS and
test them based on amplitude.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod:synch: add separators to improve logs

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: add failed tests count

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synch: add vcxo switch testing

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: add remote logging

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: add remote logging

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: update fru command to match latest version

fru tools now use command line arguments for offset
if the file is larger than standard 256 bytes. The fru-dump
command was updated accordingly.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
production: initial commit

Re-organize test folder structure to avoid code
and file duplication, add new folder for som2.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: generalize setup+production scripts

Moved necessary folders and files in main_tests, and
added appropriate parameters to the setup and production
scripts in order to work with the chosen setup. These
changes eliminate a grreat part of the duplicated code
and make the process of adding new tests easier.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: adrv9361 prod test initial commit

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: adrv9361: increase wait time for otg

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: adrv9361: add fpga loopback test

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: adrv9361: add writing image to flash

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: synchrona: update eeprom image

Update eeprom image to latest available version and
configured to choose USB boot first if available.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: adrv9361: check tty port before testing UART

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: adrv9361: check tty port before testing UART

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: tests common: add support for remote logging

Added remote logging for all boards in main_tests

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: tests common: add support for remote logging

Added remote logging for all boards in main_tests

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod:synch: rebase with latest branch

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

tachicialex commented:
Can you please upload the rpi-ad9545-hmc7044 devicetree's sources (all 3 of them)? Also, maybe instead of using uploaded binaries you can use [dtc](https://manpages.debian.org/jessie/device-tree-compiler/dtc.1.en.html) to compile them on the fly.

ranechita committed:
prod: synchrona: adapt tests to newest synchrona image

The latest modifications in the latest version of
drivers, kuiper and synchrona devicetree, led to a
series of modifications in the test scripts. The
devicetrees have been updated to match the latest
version, and a few minor modifications regarding
removing overlays and the static ip have been
added.

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
production-all: install and configure telemetry in setup_env

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: setup env TBDs

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

constmonica committed:
Cleaned-up file: enabled python bindings, added board function area

constmonica committed:
added production.sh files to each test folder, cleaned production.sh

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
created a script for automation of test creation

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
created a file containing the board names

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Added disable sudo function

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Modified sed command to append new board name to supported_boards.sh

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Fixed typos in production.sh files

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Added code line to change the hostname of the pi in hosts file.

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Changes to $SCRIPT_DIR and added exit 1; after handle error state


Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Moved functions from lib/production.sh to solve circular
reference error


Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Moved functions to utils.sh to solve circular reference error


Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Fixed a typo in +STEPS

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Changed wrapper script description

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

ranechita committed:
talise: update test files for talise prod to latest

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

ranechita committed:
prod: Add test files for fmcomms2/3

Signed-off-by: Ramona Alexandra Nechita <ramona.nechita@analog.com>

constmonica committed:
Added a code line to make production.sh files executable.

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Fixed typos

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Fixed typos and changed "serial" from local variable

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Fixed path errors in LOGDIR and $SCRIPTDIR typos

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Added MODE for the case command

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Merge branch 'reorganize_tests' of https://github.com/analogdevicesinc/production-tests into reorganize_tests

constmonica committed:
Updated talise and fmcomms8 test files.

Signed-off-by: constmonica <monica.constandachi@analog.com>

constmonica committed:
Added fmcomms2-3 support

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Added fmcomms2-3 prod test files

Signed-off-by: Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Updated test menu

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Add production testing files for 60GHz-wireless connector

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Added support for 60GHz-conn production test


Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Fix the disable of wifi after running setup_env.sh


Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Fixed some typos


Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Add test files for SWIOT production test

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Added support for SWIOT production test.

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Fixed a typo

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Added max-arduino production test files

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Added support for max=arduino prod test.

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Fixed code typos.

constmonica committed:
Added max-arduino  requirements to setup_board fct.

constmonica committed:
Update test files

constmonica committed:
Fixed a variable typo.

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Add EV-charger test files

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Add EV charger test support

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Update max-arduino test scripts.

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Add support for usb-to-t1l prod test.

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
USB-TO-T1L production test files

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Update test files for max-arduino rev E.


Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Add support for GMSL716MIPI test

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Add gmsl716mipi test files.

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Add support for apard-pfwd board

Signed-off-by:Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
main_tests/apard-pfwd: add test files

Signed-off-by:Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Add support for rpiplcpse board test.

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
main_tests/rpiplcpse: add test files

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Add support for apard-spoe board test

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
main_tests/apard-spoe: add test files

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
/main_tests: add eth2gmsl test files


Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
Merge branch 'reorganize_tests' of https://github.com/analogdevicesinc/production-tests into reorganize_tests

constmonica committed:
main_tests/max-arduino: add missing files

Signed-off-by:Monica Constandachi<monica.constandachi@analog.com>

constmonica committed:
main_tests: add helpkit test files

Signed-off-by: Monica Constandachi <monica.constandachi@analog.com>

constmonica committed:
fix typo

Signed-off-by: Monica Constandachi <monica.constandachi@analog.com>

constmonica committed:
add support for pqm test

constmonica committed:
/main-tests: add test files for pqm board


Signed-off-by: Monica Constanddachi <monica.constandachi@analog.com>

constmonica committed:
update firmware scripts

Signed-off-by: Monica Constandachi <monica.constandachi@analog.com>

trupples committed:
supported_boards.sh: one board per line

Aids in easier diffing & conflict resolution between revisions.

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
main_tests: Set executable flag on all production*.sh

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Add ADRD3161 test procedure

Setup:
* Install Docker
* Build & install Openocd with support for MAX32xxx ICs
* Install Ubltools (requires GITHUB_TOKEN variable to be set)
* Download zephyr toolchain
* Build docker image with ROS 2 automated test (see adrd3161_tests)

Runtime:
* Use ublcli to bootstrap TMC9660
* Use zephyr gdb to modify firmware with serial number
* Use openocd to write MAX32662 firmware
* Run docker container with tests and report any errors

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
Add python3-virtualenv to apt_prereqs

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

trupples committed:
adrd3161: Fix TMC bootstrap issues

* Flush out initial UART garbage from signal bounce when connecting the probe
* Exit bootstrap mode on a successful OTP write

Signed-off-by: Ioan Dragomir <ioan.dragomir@analog.com>

plescaevelyn committed:
main_tests: Update environment setup steps

- Update telemetry instalation and add cryptography version update to avoid dependency version errors
- Change default Python version to being system appropriate
- Update libiio installation steps

Signed-off-by: Evelyn Plesca <evelyn-iulia.plesca@analog.com>

cristina-suteu committed:
main_tests: add ADRD5161 prod test

Signed-off-by: Cristina Suteu <cristina.suteu@analog.com>

plescaevelyn committed:
main_tests: Make setup_env work on both Kuiper 1 and 2

- Modify pip installation test based on debian versions
- Setup network using DHCP or Network Manager based on the Kuiper version used to match distro preferences

Signed-off-by: Evelyn Plesca <evelyn-iulia.plesca@analog.com>

constmonica committed:
gmsl716mipi/cam_setup.sh: update for new board revisions

Signed-off-by:Monica Constandachi <monica.constandachi@analog.com>

plescaevelyn committed:
main_tests/adrd4161: Add LED and IMU tests

- Add config files used in firmware application
- Apply firmware and test RX TX LED blink
- Apply testing firmware that allows slcan to work
- Initialize CAN protocol, test CAN dump, CAN RX and TX
- Test whether LED set on GPIO 24 has turned ON
- Test the presence of measurements on the ADIS16470

Signed-off-by: Evelyn Plesca <evelyn-iulia.plesca@analog.com>

plescaevelyn committed:
main_tests: Add ADRD4161 in supported boards list

Signed-off-by: Evelyn Plesca <evelyn-iulia.plesca@analog.com>

plescaevelyn committed:
main_tests/adrd4161: Modify LED names

Signed-off-by: Evelyn Plesca <evelyn-iulia.plesca@analog.com>

cristina-suteu committed:
adrd5161: update fw

Signed-off-by: Cristina Suteu <cristina.suteu@analog.com>

plescaevelyn committed:
main_tests: Update ADRD4161 setup steps

- Automatically add overlays in the /boot/config.txt file
- Update open ocd setup steps by adding necessary dependency in the installation steps
- Fix misspelling

Signed-off-by: Evelyn Plesca <evelyn-iulia.plesca@analog.com>

constmonica committed:
main_tests: fix typos

Signed-off-by: Monica Constandachi <monica.constandachi@analog.com>

constmonica committed:
main_tests: add fmcdaq3 test files
Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
main_tests/swiot: add the latest ip

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>

constmonica committed:
swiot: fix CRLF endings

Signed-off-by: constmonica <Monica.Constandachi@analog.com>

constmonica committed:
.github: add shellcheck workflow

Signed-off-by: Monica Constandachi <Monica.Constandachi@analog.com>
